glommio/src/executor/mod.rs

```diff
@@ -1060,7 +1155,31 @@ impl LocalExecutor {
                 config.record_io_latencies,
                 config.thread_pool_placement,
             )?),
-        })
+            stall_detector: None,
+        };
+        exec.detect_stalls(config.detect_stalls)?;
+
+        Ok(exec)
+    }
+
+    /// Enable or disable task stall detection at runtime
+    ///
+    /// # Examples
+    /// ```
+    /// use glommio::{DefaultStallDetectionHandler, LocalExecutor};
+    ///
+    /// let local_ex =
+    ///     LocalExecutor::default().detect_stalls(Some(Box::new(DefaultStallDetectionHandler {})));
+    /// ```
+    pub fn detect_stalls(
+        &mut self,
+        handler: Option<Box<dyn stall::StallDetectionHandler + 'static>>,
+    ) -> Result<()> {
+        self.stall_detector = match handler {
+            Some(handler) => Some(StallDetector::new(self.id, handler)?),
+            None => None,
+        };
+        Ok(())
```

Comment:
Can you add one final test that ensures this is safe for `pub` visibility? i.e., that we can safely disable it from outside the reactor? 

I'm asking because while user code is touching this, the stall detector is actively used, and a timer exists. I want to ensure that disabling it doesn't result in anything weird. 

Two little tests should do:
* A test with a cooperative task that disables the stall detector mid-course -> assert that we don't record anything;
* A test with an unyielding task that disables the stall detector mid-course -> assert that we don't produce a trace (or that we do one last time, depending on what behavior is easier to implement).

Reply:
I removed the export for now.